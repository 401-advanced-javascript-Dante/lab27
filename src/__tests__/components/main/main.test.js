import React from 'react';
import {shallow , mount} from 'enzyme';
import renderer from 'react-test-renderer';
import Main from '../../../components/main/main.js';


describe('<Main />', () => {

  it('already exist in our app' , () => {
    let test = shallow(<Main/>);
    expect(test.find('button').exists()).toBeTruthy();
  });

  it('Assert state changes properly', () => {
    let test = mount(<Main/>);
    let addButton= test.find('#left');
    addButton.simulate('click');
    expect(test.state('counter')).toBe(1);
  });

  it('Assert that state is being transferred to the DOM', () => {
    let test = mount(<Main/>);
    let button= test.find('#right');
    button.simulate('click');
    expect(test.find('#number').text()).toBe('-1');
  });

  it('Assert DOM stability via snapshot', ()=> {
    let test = renderer.create(<Main />);
    expect(test).toMatchSnapshot();
  });



});