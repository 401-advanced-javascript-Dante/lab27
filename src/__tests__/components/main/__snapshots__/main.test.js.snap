// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Assert DOM stability via snapshot 1`] = `
<section>
  <div
    id="number"
  >
    0
  </div>
  <div>
    <button
      id="left"
      onClick={[Function]}
    >
      +
    </button>
    <button
      id="right"
      onClick={[Function]}
    >
      -
    </button>
  </div>
</section>
`;
